java/src/main/native/src/maps_column_view.cu

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/detail/replace.hpp>
+#include <cudf/lists/contains.hpp>
+#include <cudf/lists/detail/contains.hpp>
+#include <cudf/lists/detail/extract.hpp>
+#include <cudf/lists/extract.hpp>
+#include <cudf/lists/lists_column_view.hpp>
+#include <cudf/scalar/scalar_factories.hpp>
+#include <cudf/types.hpp>
+#include <maps_column_view.hpp>
+#include <rmm/exec_policy.hpp>
+
+namespace cudf::jni {
+
+namespace {
+column_view make_lists(column_view const &lists_child, lists_column_view const &lists_of_structs) {
+  return column_view{data_type{type_id::LIST},
+                     lists_of_structs.size(),
+                     nullptr,
+                     lists_of_structs.null_mask(),
+                     lists_of_structs.null_count(),
+                     lists_of_structs.offset(),
+                     {lists_of_structs.offsets(), lists_child}};
+}
+} // namespace
+
+maps_column_view::maps_column_view(lists_column_view const &lists_of_structs,
+                                   rmm::cuda_stream_view stream)
+    : keys_{make_lists(lists_of_structs.child().child(0), lists_of_structs)},
+      values_{make_lists(lists_of_structs.child().child(1), lists_of_structs)} {
+  auto const structs = lists_of_structs.child();
+  CUDF_EXPECTS(structs.type().id() == type_id::STRUCT,
+               "maps_column_view input must have exactly 1 child (STRUCT) column.");
+}
+
+template <typename KeyT>
+std::unique_ptr<column> get_values_for_impl(maps_column_view const &maps_view,
+                                            KeyT const &lookup_keys, rmm::cuda_stream_view stream,
+                                            rmm::mr::device_memory_resource *mr) {
+  auto const keys_ = maps_view.keys();
+  auto const values_ = maps_view.values();
+  CUDF_EXPECTS(lookup_keys.type().id() == keys_.child().type().id(),
+               "Lookup keys must have the same type as the keys of the map column.");
+  auto key_indices =
+      lists::detail::index_of(keys_, lookup_keys, lists::duplicate_find_option::FIND_LAST, stream);
+  auto constexpr absent_offset = size_type{-1};
+  auto constexpr nullity_offset = std::numeric_limits<size_type>::max();
```

Comment:
I think this should use`std::numeric_limits<size_type>::min()` instead.  This is similar to the gather case, where an index that is allowed to be negative can be used. Theoretically a list column could have a single row with a child column with the maximum number of elements in that single row, but the min size_type value is an invalid index even when treated as an index from the end.

Reply:
> This is similar to the gather case, where an index that is allowed to be negative can be used.

It turns out that using `numeric_limits::min()` and `numeric_limit::max()` are both correct in this case.

If a list has `MAX_INT` entries, its valid (positive) indices are in the range `[0, MAX_INT - 1]`. Since `MAX_INT` falls outside that range, it is nullified.

Similarly, a list with `MAX_INT` entries would have valid negative indices in the range `[MIN_INT + 1, -1]`. Thus, `MIN_INT` would also cause `segmented_gather()` to nullify.

Would you prefer that we use `min()` instead of `max()`?

Reply:
min() would be consistent with the other cases that are using an invalid index to generate nulls.